x/epochstorage/keeper/epoch_start.go

```diff
@@ -20,9 +26,96 @@ func (k Keeper) EpochStart(ctx sdk.Context) {
 
 	k.SetEpochDetailsStart(ctx, block)
 
-	k.StoreCurrentEpochStakeStorage(ctx, block)
+	k.StoreCurrentStakeEntries(ctx, block)
 
 	k.UpdateEarliestEpochstart(ctx)
 
 	k.RemoveOldEpochData(ctx)
 }
+
+// StoreCurrentStakeEntries store the current stake entries in the epoch-prefixed stake entries store
+func (k Keeper) StoreCurrentStakeEntries(ctx sdk.Context, epoch uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.StakeEntriesCurrentPrefix)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{}) // Get an iterator with no prefix to iterate over all keys
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var currentEntry types.StakeEntry
+		k.cdc.MustUnmarshal(iterator.Value(), &currentEntry)
+		k.SetStakeEntry(ctx, epoch, currentEntry)
```

Comment:
maybe we can do without marshaling since we dont change the entry itself? just adding with a different key

Reply:
I can't since the new key needs data from the unmarshalled stake entry: chain, effective stake and provider address